bn_testing/dags.py

```diff
@@ -120,6 +124,7 @@ def sample(self, n):
 
         df = pd.DataFrame()
 
+        # this has to start at the sources of the graph, right ?
```

Comment:
this has to start at the sources of the graph, right ?

Reply:
Exactly, but as the nodes are topologically ordered, all required nodes for a given nodes are already generated. If a node is not in `self.models`, it doesn't has parents and hence its independent. Does that make sense?

Reply:
yes, thanks for the explanation!